### PR TITLE
Task/csi 4683 mediator interface

### PR DESCRIPTION
**Pull request by ELENAGER**

### CODE REVIEW

File: controllers/servers/csi/addons_server.py [hunk 1]
```diff
@@ -131,26 +121,48 @@ def DemoteVolume(self, request, context):
...
+        other_volume_id_info = utils.get_volume_id_info(request.replication_id)
+        other_volume_internal_id = other_volume_id_info.ids.internal_id
+
+        other_system_id = request.parameters.get(servers_settings.PARAMETERS_SYSTEM_ID)
+        copy_type = request.parameters.get(servers_settings.PARAMETERS_COPY_TYPE, REPLICATION_DEFAULT_COPY_TYPE)
+        return ReplicationRequest(volume_internal_id=volume_internal_id,
+                                  other_volume_internal_id=other_volume_internal_id,
+                                  other_system_id=other_system_id,
+                                  copy_type=copy_type,
+                                  replication_type=replication_type)
```
Comment (ArbelNathan) [inline controllers/servers/csi/addons_server.py:168]: please move this to utils

File: controllers/servers/csi/addons_server.py [hunk 2]
```diff
@@ -18,62 +20,54 @@ class ReplicationControllerServicer(pb2_grpc.ControllerServicer):
...
         volume_id_info = utils.get_volume_id_info(request.volume_id)
         volume_id = volume_id_info.ids.uid
-        volume_internal_id = volume_id_info.ids.internal_id
-
-        other_volume_id_info = utils.get_volume_id_info(request.replication_id)
-        other_volume_internal_id = other_volume_id_info.ids.internal_id
-
-        other_system_id = request.parameters.get(servers_settings.PARAMETERS_SYSTEM_ID)
-        copy_type = request.parameters.get(servers_settings.PARAMETERS_COPY_TYPE, REPLICATION_DEFAULT_COPY_TYPE)
+        rep_request = self._generate_replication_request(request, replication_type)
```
Comment (ArbelNathan) [inline controllers/servers/csi/addons_server.py:28]: `replication_request`? same in all methods

Comment (ELENAGER) [reply controllers/servers/csi/addons_server.py:28]: This replication request is used inside get_replication function of mediator. And it is the same

Comment (ArbelNathan) [reply controllers/servers/csi/addons_server.py]: I think we should change its name to `replication_request`

Comment (ArbelNathan) [reply controllers/servers/csi/addons_server.py]: same in all methods

Comment (ELENAGER) [reply controllers/servers/csi/addons_server.py]: Done

File: controllers/array_action/settings.py [hunk 3]
```diff
@@ -5,6 +5,8 @@
 ISCSI_CONNECTIVITY_TYPE = config.connectivity_type.iscsi
 REPLICATION_COPY_TYPE_SYNC = "sync"
 REPLICATION_COPY_TYPE_ASYNC = "async"
+REPLICATION_TYPE_MIRROR = "mirror"
+REPLICATION_TYPE_EAR = "ear"
```
Comment (ArbelNathan) [inline controllers/array_action/settings.py:9]: where is the use for this?

Comment (ELENAGER) [reply controllers/array_action/settings.py:9]: In validate_addons_request and _get_replication_type

Comment (ArbelNathan) [reply controllers/array_action/settings.py:9]: I can find REPLICATION_TYPE_EAR anywhere..

Comment (ELENAGER) [reply controllers/array_action/settings.py:9]: You are right, I just added it for symmetry. Do you want me to remove it?

Comment (ELENAGER) [reply controllers/array_action/settings.py:9]: Sorry, my mistake. I am using it here:
    def _get_replication_type(request):
        if servers_settings.PARAMETERS_REPLICATION_POLICY in request.parameters:
            replication_type = array_settings.REPLICATION_TYPE_EAR
        else:
            replication_type = array_settings.REPLICATION_TYPE_MIRROR

        logger.info("replication type is {}".format(replication_type))
        return replication_type

**Review by ArbelNathan — APPROVED** [review]
LGTM

**Review by matancarmeli7 — APPROVED** [review]
LGTM

**Pull request merged**